### DISCUSSION TITLE
2024-03-15 - The TinaCMS Customer Experience - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by Dhruv-0987** [SSWConsulting/SSW.Rules.Content#8137] [Chewing the Fat]
Hey SSWers,

We have purchased Tina and need a roadmap to make its customer journey better.

Video: https://youtu.be/uJde9cz24ww

Let the ideas 💡 flow.

**Comment by calumjs** [comment]
I'd like to know what the AI roadmap looks like. 
Ideas off the top of my head:
- Content generation assistant
- Auto content review (style guide enforcement, fuzzy duplicate detection)
- Multilingual auto translation
- Built-in end user chatbot with access to your cms data (e.g. rules gpt would come included for an extra fee or something)

**Comment by PothieuG** [comment]
I imagine we could internationalize the application:

- TinaCloud, the website, etc., translated into languages based on the user's location.
- having the choice between the most spoken/used languages in the dev world, i.e. : English, Chinese, Russian, German, Spanish, and of course, French 🥖🐓!

**Reply by leahy268** [reply]
My order really was:
1. Dev Templates: Not being able to successfully create something straight away is bad
2 Hosting on TinaCloud and Website Experience - I saw as those as same level of priority.   SEO done yesterday.  
4. Customer Support - Especially when it is big that's hugely important

**Comment by Freego1783** [comment]
I think that fixingthe Dev Templates and hosting on TinaCloud should be a priority. The kind of experience that was presented to us in the video, combined with the lack of direct support, is going to discourage a lot of people, which is a shame.

**Comment by AlexRothwell** [comment]
The work done on identifying the pain points and putting together the roadmap so far looks great, well done all involved! From my initial look at it all, I'm quite happy with the plan - the issues identified are important and the fixes are achievable

**Comment by bahjatalaadelssw** [comment]
Get the UX sorted first, because if the navigation and experience is painful people won't be willing to learn more.

**Comment by ozdemir-mehmet-cer** [comment]
Presentation looked very good overall

**Comment by stephanfakossw** [comment]
Great video explaining what we need to improve.

**Comment by BrookJeynes** [comment]
Dev templates should definitely be a priority. If a dev can't mess around with the product it then becomes a very hard sell

**Comment by ricksu978** [comment]
My roadmap priority
```
1. Website Experience (first impression)
2. Dev Template       (first hands on experience)
3. Hosting TinaCloud  (important)
4. Customer Support   (still important, but if first 3 items are done properly, we'll get less support enquiries)
```

**Comment by 2fernandez** [comment]
Customer journeys do not typically start on the website.

The video starts on the second step of a customer journey, the "Explore" phase—which is also where consumers get their first impressions.

And then jumps right into the "Onboarding", almost the last step of a customer journey.

**Comment by MarcusIrmscherSSW** [comment]
Great video Levi, look forward to seeing more from you. 👍👍

**Comment by joshbermanssw** [comment]
Great video Levi, Kiki and Matt! I personally thought that the functionality of setting up Tina (and second to that, hosting Tina in cloud) is integral. Once the foundations work, UX and website can follow.

**Comment by YalalCakirSSW** [comment]
I think roadmap should be like

1. `Website experience` 
2. `Dev Templates`
3. `Hosting on TinaCloud`
4. `Customer Support` 

You need customers to give a customer support, therefore I think user experience is the most important one with the Dev Templates because these are things represented the product, an having the good features is the key.

**Comment by uly1** [comment]
Getting this right is absolutely vital. It almost doesn't matter how good the product is if most people get turned off by the experience of trying to see what it os or how to use it.

**Comment by Calinator444** [comment]
Personally I think the demo on the Tina CMS website is already quite decent. From the sound of things the developer experience could be improved. I think it's important for developers to have a working demo when they use the template so that they have a good reference point before they start adding changes later.

Here's a link to the current tina demo: https://quick-edit-demo.vercel.app/admin#/~

I also think there's some room for improvement in the SEO space. Tina CMS is easy to find on google if I know it exists, but it's further along in the search results if I'm just looking for a CMS.

<img width="575" alt="image" src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/65635198/c6d8c145-5ee3-4e34-8f10-e5e8e6f2966a">

**Figure**: **search results for "CMS"**

**Comment by SamBWagner** [comment]
I looked at the site, it looks great. I think that making sure people can use it and get support is the most important thing so

1. It works (if TinaCMS is in a non-functional state then it's literally worthless).
2. The onboarding of it. If people find it hard to use then it's also going to die.
  - Good doco
  - Simple usage
3. Support. If people don't feel like they can get their feedback heard then the project will lose it's users.

Everything else seems superfluous by comparison.

**Comment by jimmy1992abc** [comment]
UX should be the most important. For any product, making something that the client wants is crucial (unless you have overwhelming domination like Apple).

**Comment by TobyChurches** [comment]
Great Video!
I think we should definitely be prioritising the templates and hosting issues before we put the time/money/effort into marketing the product. First impressions really do count and if someone tries to use TinaCMS for the first time and it fails, chances are we are not going to win that user even if we do have a bunch of great features. Users want a solid product, especially when they are paying good money for it

**Comment by AntPolkanov** [comment]
Great video, watched it all!
In my opinion the priority should be:

1. Dev templates
2. Hosting
3. Website Experience
4. Customer support

Considering that 95% of our users are developers, ensuring a smooth developer experience is crucial. 
While the website could use some enhancements, it's currently sufficient (and I like gifs 😜).
The need for customer support can be partially mitigated by good onboarding experience and docs. Also would be nice to have a few tutorials to help people to get started.

**Comment by Marxoz** [comment]
Awesome video Levi and team!

Getting those templates to work should be highest priority, as well as having a demo site / video. You want customers to be able to try it out and experience its awesomeness. 

Next for me would be customer service. Once they have tried it out, they would most certainly have some questions about the product, and we will want to jump on it like a fly on a turd. If they are asking questions, they are interested, and we can offer them a free live demo. 

Also, throwing a chatbot on there might be easier than redoing the website for now.

**Comment by bettybondoc** [comment]
Great presentation! ✨

The 'Dev Templates' and 'Hosting on TinaCloud' are actually an extension of the 'Website Experience'. Once a user is on the website, they would definitely want to trial the product and if they can't get it to work, they would just blame the overall experience so getting the templates and hosting sorted is definitely the priority.

In saying that, fixing the website experience and the dev templates/hosting issues can happen simultaneously because the first one would initially be more marketing/designer heavy and the second one would be dev heavy. So, it doesn't have to be one after the other because this requires different skillsets.

**Comment by tkapa** [comment]
Good video :), likely could have been split up into parts.

Lots of people are apparently having problems with getting started on the Dev Templates, so that should be the priority here. 

I'm more excited to jump into the meat on this.

**Comment by jeoffreyfischer** [comment]
Great roadmap! It covers 90% of Tina's current issues

**Comment by matt-goldman** [comment]
Given that our vector is developers we should focus on the DX. Make sure the templates work, make sure the hosting works, make sure the website is good, make sure the support is in place.

I think we're missing Community Engagement from the roadmap. The current criticisms around Discord are valid, but again given the developer vector, this is an important part. But we need to expand this and think about how we can get devs talking about this, how we can get them thinking it's cool. Then build up from there to start thinking about what the arguments they can include in a business case would be (once a developer likes something, they then have to pitch it to the purse strings holders/bean counters).

So things like privacy, security, and cost should be first class citizens of the product and easily surfaced information on the website.


Edit to add: questions 4 - 7 should have been one question:

<img src="https://github.com/SSWConsulting/SSW.Rules.Content/assets/19944129/109bdef4-9ff8-4caf-8d3c-5a75cd584538" width="600" />

**Comment by alexleessw** [comment]
Great presentation Levi!

**Comment by AshAnil** [comment]
Nice video showing the pain and solutions.

**Comment by cvionnet** [comment]
Great video 👍!
My opinion is that we should prioritize addressing the pain points involved in setting up the product, as 95% of the users are developers. It's not that the website is perfect; it surely can be improved, but it is not that bad.

1. Developer templates
2. Hosting
3. Website Experience
4. Customer Support

One step that is missing is how the acquisition by SSW, a private company, will be perceived by the current users and the open-source community. I think it is important to communicate about this point to avoid any misunderstandings that could lead the community to reject the project.

**Comment by AttackOnMorty** [comment]
Good video. Fixing dev templates is a quick win.

**Comment by awaltos** [comment]
Great video - great to see lots of work has gone into researching and plainly laying out the pain points. 

I think that the identified buggy behaviour should be resolved asap - but these are more like issues than roadmap items. Dev experience is also key, optimising things like local build time and indexing to give devs a real wow experience on the first impression. As someone who would try different products like this before landing on one, it can take just one bad experience to drop a product.

For me, website is overall a good experience at the moment, I like its simplicity and think that adding carousels and videos on landing page could add clutter. I like the idea of a 'Try now' CTA. Sign up could def be streamlined but the wizard icon is weird. Website has some accessibility issues which seem to be trivial to solve. Some of the Benefits and Comparisons pages are very similar - could be consolidated, e.g., Comparisons could be done in one table. SEO is very important but should probably be done once the initial experience issues are sorted out.

Ticket support is important, but I agree that it can be a later priority whilst issues/day are low. It's an open-source product so people understand what to expect especially the free-tier ppl. However, the paid-tier ppl should get some more love for sure.

Good point by @cvionnet about the open-source bit. https://github.com/SSWConsulting/SSW.Rules.Content/discussions/8137#discussioncomment-8792456

I'm excited by this acquisition and look forward to seeing the uplift and hopefully being able to contribute 😄

**Comment by ncn-ssw** [comment]
My prioritizations:
1. Dev templates
2. Website experience
3. Hosting on TinaCloud
4. Customer support
My concern with the demo instance proposal in "Pain 2 - Hosting on TinaCloud" is securing so that it doesn't become a hosting site for inappropriate material.

**Comment by jakebayliss** [comment]
I think getting it set up with a good development experience is priotity 1!

**Comment by Harry-Ross** [comment]
I agree that the focus should be on the GitHub-backed CMS part, they offer other options but it basically means we have to support using a database adapter (i.e. using SQLite) instead of focusing on the reason why people are actually using Tina - for the GitHub support. 

Advertising for Tina is also super important - I see a lot of ads for Sanity (one of Tina's main competitors), so I imagine its important to focus on name recognition for Tina in the way that Contenful, Builder.IO and Sanity have name recognition. Something also to note is that the reason why Builder.IO has name recognition is due to their open source frameworks they've created like Quik (JS framework), Partytown (web worker library) and many others, so that serves as its own form of advertising. 

I disagree with the point made about Discord - this is the main place that many open source projects keep their communities engaged. It's important to have a communal space where people can discuss the product and get quick help if needed. See Next.js and Astro's Discord servers, where a big part of why a person would choose the product is the community support. 

From the dev experience side - Next.js App Router should be #1 priority - we should not have the only option of using Tina to be using the old way of building a Next.js app.

**Comment by Ben0189** [comment]
Good video showing the pain,  website is where we reach out to potential users the most. if it doesn't attract users, they won't even get to the other pain (that means solving the other pain will not be valuable). 

Marketing should also be equally important to improving the products

**Comment by sethdaily** [comment]
Website experience is #1 - if it's not good they won't be willing to go any further

**Comment by chrisschultzssw** [comment]
I'm wondering who our target market is - if 95% of current customers(?) are devs, there's probably a good reason for that.
Let's focus on the people the product is currently reaching before trying to appeal to a wider market.

**Comment by jaydenalchin** [comment]
Interesting ideas! Website UX matters a lot - we only get one chance at a first impression as Levi mentioned. 

A lot of the listed UX pain points and proposed solutions can & should be applied to the SSW landing page too 🥺

**Comment by steven0x51** [comment]
Good video! 
My prioritizations:

 1. Website experience
 2. Dev templates
 3. Hosting on TinaCloud
 4. Customer support

**Comment by tomek-i** [comment]
It is actually a common occurrence that I have found with "starter" templates not working at all. When we did the Leonardo.ai challenge the TsED template did not work and the documentation was very poor. 

If I am on the lookout for a product and basically trying stuff out to compare them against each other and their "starter project/template" is not working then that product is dead ☠️ to me and I move on.

**Comment by amankumarrr** [comment]
My prioritization:

1. Dev templates
2. Customer support
3. Hosting on TinaCloud
4. Website experience

I agree that ensuring a positive website experience is crucial, but prioritizing support for our existing clients is equally essential. By delivering the best possible experience to those already using Tina Cloud, we can cultivate positive word-of-mouth marketing. While a good website experience can attract new audiences, offering a buggy template and inadequate customer service could deter potential clients.

The pain point for the video could be framed as something like:
"Why should we use a CMS, and more importantly, why Tina?"

**Comment by JackDevAU** [comment]
Awesome video!! 

Excited to see Tina grow and make it truly the best git based CMS. 

I've tried out TinaCMS a few times and I found that the docs are what's lacking the most. This **needs** **to be improved!** 
Having good docs makes dev life so much easier!

**Comment by CoffeeBugHub** [comment]
My roadmap priorities are: 

1. Dev templates
2. Hosting on TinaCloud
3. Website experience
4. Customer support

Also, I'm curious about the significant efforts being made to address the needs of non-developers. Who will be the primary audience for this product?

**Comment by MattParkerDev** [comment]
Seo is definitely important for bringing in new customers and increasing adoption. I agree with all of the next steps in the video.

**Comment by camillars** [comment]
Great job, Levi! 😊

My priority order would be:
1. website experience (if the website UX is not good enough, they won't want to learn more)
2. customer support (with no or very little support, and this includes answering questions before they purchase the product, the clients won't want to stay)
3. hosting on TinaCloud (where ideally profit will come from)
4. Dev templates (devs need support, but if we offer a better UI with pre-existing blocks (like WordPress) and content to add to the website for both devs and non-devs, we won't need this so urgently)

**Comment by Levijj22** [comment]
Great to read everyone's opinions and thoughts
Let's build the best CMS!

**Comment by BenDunkerleySSW** [comment]
Great video Levi. I think the website is the number one priority - it is the port key. Existing customers are important, but they are just the beginning. Tina has a big opportunity to grow, and I think the website will make the largest dent in that (alongside generally more exposure)

**Comment by christoment** [comment]
There's a lot of things we can improve on Tina - I think customer experience plays a huge role on getting more people to use this.
We should also aim to deliver couple of features from TinaCMS and TinaCloud as well (e.g. AI powered alt text for media manager, easy SSO) to entice more customer 😀

**Comment by PennyWalker** [comment]
I think the customer experience will be key here. My first instinct was to get the website looking better and more user-friendly, but that will drive more traffic to the 'try before you buy stage' so I think we need to get those onboarding templates fixed first so the customer has a great experience. Then make a great user-friendly site that will attract more customers - this will support the marketing down the track as people need to understand the product straight away in the discovery stage.

**Comment by jimmidier** [comment]
Great video! As a dev I'd love to see the docs improved. The current is still too generic and lacks helps for troubleshooting problems

**Comment by danielmackay** [comment]
I love this idea!  Very similar to what the ASP.NET Core templates and other templates do.  Seems like a no-brainer.

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/2636640/5de787b2-3911-4af0-bcba-1d92bb0a05f3)

I found it hard to read the on screen text during Kiki's demo

I struggle to see the value in TinaCloud.  It's trivial to get a FE running on Vercel.  What problem is TinaCloud solving?

**Comment by Aibono1225** [comment]
Great presentation, it really points out a lot of improvements that we can make with Tina

**Comment by KristenHu** [comment]
Great video, my priority order for the roadmap would be:
1. Dev templates
2. Customer support
3. Hosting on TinaCloud
4. Website experience

Happy to see that we're improving TinaCMS🎉

**Comment by miichaelsmedley** [comment]
Sell it first - Website experience
make it look good - dev templates
Host it - Hosting on Tina Cloud
Support it - Customer support

**Comment by babakamyljanovssw** [comment]
Great video Levi!

My priorities for roadmap are:

1. Dev templates
2. Website experience
3. Hosting on TinaCloud
4. Customer support

Fixing dev templates first would be huge step towards to product goal.

**Comment by synnoveisabel** [comment]
Great video Levi! I think all the pain points are important to solve

**Comment by ozair-ashfaque** [comment]
1. `Website experience`
2. `Dev Templates`
3. `Hosting on TinaCloud`
4. `Customer Support`

**Comment by drwharris** [comment]
Set yourself up for customer success first, because you only get 1 first impression right?

1. Customer Support
2. Website Experience

3 & 4 become about what you want to sell more of...cloud or non-cloud?

**Comment by isaaclombardssw** [comment]
Levi's presentation is very clear in linking up pain points and solutions. The website and then the template issues would be my first priorities. Super exciting project!

**Comment by vladislav-kir** [comment]
Great video from Levi, Kiki and Matt!

**Comment by jackreimers** [comment]
Great video that highlights the important issues that need to be addressed with Tina

**Comment by tino-liu** [comment]
Being able to generate a workable Dev Template is a must as it provides initial impression on the user experience. I can see it being a dealbreaker if it doesn't work. So that definitely would be top priority.
